provider/inspec.rb

```diff
@@ -73,8 +74,81 @@ def generate_typed_array(data, prop) end
 
     def emit_resourceref_object(data) end
 
-    def emit_nested_object(data) end
-
     def generate_network_datas(data, object) end
+
+    def emit_nested_object(data)
```

Comment:
No specific need for these to be in separate files except for a separation of concerns. I like separating them out stylistically, especially if multiple resources are using the same nested object.

Yeah, we are building out the class system so that we can reference nested objects in a sane way. Also so that nested objects could theoretically have custom methods on them for testing.